geopandas/sindex.py

```diff
@@ -293,6 +295,17 @@ def __init__(self, geometry):
             # store geometries, including empty geometries for user access
             self.geometries = geometry.copy()
 
+        @property
+        def valid_query_predicates(self):
+            """Returns valid predicates for this spatial index.
+
+            Returns
+            -------
+            set
+                Set of valid predicates for this spatial index.
+            """
+            return pygeos.strtree.VALID_PREDICATES | set([None])
```

Comment:
@brendan-ward Do you think that maybe `None` should be added within pygeos?

Reply:
I think we could rather leave out `None` from the enumeration, and don't check the value if it is None (assuming that this is always supported)

Reply:
Hmm. I felt that `valid_query_predicates` should return a set of all values that can be passed as `predicate=value` that will _not_ raise an error. Since `predicate=None` does not raise an error, I'd think it should be included in this set. That's more general than assuming `None` is a special case for all spatial indexes.

Reply:
I think we'll need a mixed strategy for the short term.  In pygeos, it would simplify and clarify the code a bit if we add `None` to the enumeration (since we're specifically testing for `None` then assigning it a value).  I'll work up a PR over there for that.  In the meantime, since that would be in the next pygeos release, the code above will be necessary.

Reply:
Luckily there's no need to sync efforts, this code will continue to work after the pygeos change, so it'll just be a future cleanup here.

Reply:
Unfortunately we can't add `None` to an enum.  Let's just go with the above code, and assume that `None` will always be valid here.